### DISCUSSION TITLE
The test script itself, which may be split across more than one YAML file

### DISCUSSION THREAD

**Discussion by TLmaK0** [artilleryio/artillery#2269] [Artillery]
[How it works](https://www.artillery.io/docs/load-testing-at-scale/aws-fargate#how-it-works) says:

```
Artillery analyzes the test script and packages up all of its dependencies into a bundle to make it available to Fargate containers executing the test. Those dependencies include:

The test script itself, which may be split across more than one YAML file
```

but I was not able to find any documentation on how to split yaml file.

Version info: 
```
VERSION INFO:

Artillery: 2.0.0-37
Node.js:   v20.6.1
OS:        linux
```

**Comment by bernardobridge** [comment]
Hi @TLmaK0 👋! Thanks for opening this. I've converted it into a discussion, as it's more a discussion topic.

If you read our [best practices on organizing code](https://www.artillery.io/docs/get-started/core-concepts#test-scripts), you'll see we recommend splitting configuration and scenario files. You can then pass a config file to the CLI by using the `--config` flag.

This is what that section mostly refers to (we'll try to make the wording in those docs be more clear too!) - if you specify that to the CLI, the Fargate command will automatically know to bundle those files.

Let us know if you have any further questions on setting up a test with Fargate!

**Reply by TLmaK0** [reply]
Thanks @bernardobridge, I don't know how to split a YAML file into multiple YAML files. YAML doen't have an import statement.

**Reply by hassy** [reply]
@TLmaK0 Could you describe your use case in more detail? If you have just one YAML file with all config and scenarios in it, you don't need to split it to run it on Fargate.

**Reply by TLmaK0** [reply]
Hi @hassy, I would like to run a flow that does a login, and use it on every scenario, or reuse flows in other scenenarios.

Something like:
```
- login
- request 1
- request 2
- request 3
- request 4

- login
- request 1
- request 2
- request 3
- request 5
```

**Reply by TLmaK0** [reply]
Also actual yaml file has 1368 lines right now

**Reply by bernardobridge** [reply]
I see @TLmaK0 - thanks for the details. There isn't currently a way to take parts of a scenario (or flows) and import them in other YAML files. However, we typically recommend keeping different scenarios separately if you don't need to run them with weights. Or alternatively, you can also use the new `--scenario-name` flag to run a single scenario from a large YAML file containing [multiple scenarios with weights](https://www.artillery.io/docs/reference/test-script#running-a-single-weighted-scenario).

---

That being said, if your login request is only being used in order to facilitate the other requests, what you might want to do is keep it in a [beforeScenario hook](https://www.artillery.io/docs/reference/test-script#running-a-single-weighted-scenario) instead. So basically you'd have this workflow:
- Login would be done in javascript code through a processor function + `beforeScenario` (or even a [before hook](https://www.artillery.io/docs/reference/test-script#before-and-after-sections) if you just need to log in once and keep a session). That can be reused in all scenarios that need it;
- You would separate different scenarios into different YAML files. You'd run them separately, so for example, on one artillery run you're testing e.g. `search a product by name`, on another scenario maybe you're testing e.g. `get a product endpoint` ;
- If those scenarios all share the same `config` portion, you can also keep that config file in its own YAML, and then use the `--config` flag (as I mentioned above);

_Note: Since the login flow would be happening in a hook, you wouldn't get Artillery metrics for it in your report. But that may be a good thing, as it keeps your reports scoped to exactly what you're testing. If you then needed to test login individually too, you could make a scenario just for logging in (using YAML this time, rather than a javascript hook)._

Does that help? 🙏

**Reply by TLmaK0** [reply]
Thanks @bernardobridge for your detailed response. I will try to implement my tests with your recomendations. :+1:

**Reply by bernardobridge** [reply]
Great @TLmaK0 ! 

Let us know if you run into any other issues. We'll also take this into consideration in the future when we try to improve this setup to provide a better way to reuse parts of scenarios 👍.
